book/src/dev/rfcs/0011-value-pool.md

```diff
@@ -0,0 +1,229 @@
+- Feature Name: 2021-03-21
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- Design PR: [ZcashFoundation/zebra#0000](https://github.com/ZcashFoundation/zebra/pull/0000)
+- Zebra Issue: [ZcashFoundation/zebra#0000](https://github.com/ZcashFoundation/zebra/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This document describes how to implement the Zcash value pools in Zebra.
+
+# Motivation
+[motivation]: #motivation
+
+Value pools are used in consensus rules in at least 2 places:
+
+- [ZIP-209](https://zips.z.cash/zip-0209) - Consensus rules to reject any block that results in negative pools. 
+- [Miner fees](https://github.com/ZcashFoundation/zebra/blob/62b3344f4d8605aad03133b8c0fa94cfb622d4a7/book/src/dev/rfcs/xxxx-block-subsidy.md#transaction-fees-calculation) - Fees collected for a block is the value pool for that block and they belong to the miner reward. 
+
+In addition, Zcashd has RPC calls that allow to get each pool on each block with `getblock()` and the value pools at current tip with `getblockchaininfo()`.
+
+# Definitions
+[definitions]: #definitions
+
+- `transparent pool` - Sum of all `tx_in` fields for transactions in the block chain, minus the sum of all `tx_out` fields for transactions in the block chain.
+- `sprout pool` - Sum of all `vpub_old` fields for transactions in the block chain, minus the sum of all `vpub_new` fields for transactions in the block chain [ZIP-209](https://zips.z.cash/zip-0209#terminology).
+- `sapling pool` - Negation of the sum of all `valueBalanceSapling` fields for transactions in the block chain [ZIP-209](https://zips.z.cash/zip-0209#terminology).
+- `orchard pool` - Negation of the sum of all `valueBalanceOrchard` fields for transactions in the block chain.
+- `transaction pool`
+- `block pool`
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Each of the 4 value pools change its value with every upcoming block. This is a state feature and Zebra handle this in the `zebra-state` crate. We propose to store the pool values among a reference block height into the disk.
+
+The proposed design goes from calculating transaction pools, summing them all to form block pools and finally summing all block pools to form the final pool value.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Changes to `zebra-chain/src/transaction.rs`
+
+We have a function to calculate each transaction pool for any transaction version:
+
+### `transparent_value_pool()`
+
+TODO
```

Comment:
Here we need to call the state service. Before making some prototype i want to make sure if it is ok for us to call state here.

Basically we will loop over the inputs getting `Outpoint` and calling `zebra_state::Request::AwaitUtxo(outpoint)`